app/src/main/kotlin/com/wire/android/ui/WireActivityViewModel.kt

```diff
@@ -53,7 +69,50 @@ class WireActivityViewModel @Inject constructor(
         }
     }
 
+    init {
+        viewModelScope.launch {
+            listenForNotificationsIfPossible()
+        }
+    }
+
+    private suspend fun listenForNotificationsIfPossible() {
+        withContext(dispatchers.io()) {
+            //TODO this intervalFlow is a temporary solution to have updated UserId,
+            // waiting for refactoring in kalium
+            intervalFlow(CHECK_USER_ID_FREQUENCY_MS)
+                .map {
+                    when (val result = currentSessionUseCase()) {
+                        is CurrentSessionResult.Success -> result.authSession.userId
+                        else -> null
+                    }
+                }
+                .distinctUntilChanged()
+                .flatMapLatest { userId ->
+                    if (userId != null) {
+                        getNotificationProvider.create(userId).getNotifications()
+                    } else {
+                        flowOf(listOf())
+                    }
+                        .scan((listOf<LocalNotificationConversation>() to listOf<LocalNotificationConversation>()))
```

Comment:
Can we get this method documented a bit better? I'm having trouble understanding the main logic 🙈

Reply:
If you're asking about the "scan" operator then yeah, kotlin flow documentation doesn't say too much about it but it's quite common operator in a RX world 😄 
![image](https://user-images.githubusercontent.com/30429749/165783473-d80c180d-c24b-474c-aa40-2031466801d8.png)
Basically, it applies "accumulator" function to each emitted value, you can set the initial value and because of that, it can be used to for instance count items or simply get the pair of "current" and "previous" value as in this particular case.